samples/subsys/usb/cdc_acm_composite/src/main.c

```diff
@@ -116,13 +116,13 @@ void main(void)
 	const struct device *dev0, *dev1;
 	uint32_t dtr = 0U;
 
-	dev0 = device_get_binding("CDC_ACM_0");
+	dev0 = DEVICE_DT_GET(DT_INST(0, zephyr_cdc_acm_uart));
 	if (!dev0) {
 		LOG_DBG("CDC_ACM_0 device not found");
 		return;
 	}
 
-	dev1 = device_get_binding("CDC_ACM_1");
+	dev1 = DEVICE_DT_GET(DT_INST(1, zephyr_cdc_acm_uart));
 	if (!dev1) {
```

Comment:
> I am working on a new macro `DEVICE_DT_GET_ALL(compat)`

I decided to stop at one layer beneath this macro here in this PR:

https://github.com/zephyrproject-rtos/zephyr/pull/37483

I wanted to avoid impacting device.h in case there are some compatibles where this might not be appropriate.

You should still be able use DT_FOREACH_STATUS_OKAY() here in this PR as follows:

```
#define DEVICE_AND_COMMA(node_id) DEVICE_DT_GET(node_id),
const struct *devices[] = { DT_FOREACH_STATUS_OKAY(zephyr_cdc_acm_uart, DEVICE_AND_COMMA) };
```